server/routes.py

```
@@ -0,0 +1,34 @@
+
+
+@router.post("/", response_model=ChatResponse)
+async def chat_endpoint(request: ChatRequest) -> ChatResponse:
+    try:
+        response_messages = await chat_service(request.model, request.messages)
+        return ChatResponse(messages=response_messages)
+    except AuthenticationError:
+        raise HTTPException(
+            status_code=500,
+            detail="OpenAI API Error: Invalid OpenAI API Key"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)